json_configs/configuration.cpp

```diff
@@ -78,12 +77,2050 @@ void vega_config::Configuration::load_config(const std::string& json_text) {
+            }
+            
+        }
+
+        
+        const boost::json::value * Enslave_value_ptr = controls_object.if_contains("Enslave");
+        if (Enslave_value_ptr != nullptr) {
+            boost::json::object Enslave_object = Enslave_value_ptr->get_object();
+            const boost::json::value * key_value_ptr = Enslave_object.if_contains("key");
+            if (key_value_ptr != nullptr) {
+                controls.Enslave.key = boost::json::value_to<std::string>(*key_value_ptr);
+            }
+            
+            const boost::json::value * modifier_value_ptr = Enslave_object.if_contains("modifier");
+            if (modifier_value_ptr != nullptr) {
+                controls.Enslave.modifier = boost::json::value_to<std::string>(*modifier_value_ptr);
+            }
+            
+        }
+
+        
+        }
+
+        
+        const boost::json::value * mouse_value_ptr = root_object.if_contains("mouse");
+        if (mouse_value_ptr != nullptr) {
+            boost::json::object mouse_object = mouse_value_ptr->get_object();
+            const boost::json::value * enabled_value_ptr = mouse_object.if_contains("enabled");
+            if (enabled_value_ptr != nullptr) {
+                mouse.enabled = boost::json::value_to<bool>(*enabled_value_ptr);
+            }
+            
+        const boost::json::value * FireKey_value_ptr = mouse_object.if_contains("FireKey");
+        if (FireKey_value_ptr != nullptr) {
+            boost::json::object FireKey_object = FireKey_value_ptr->get_object();
+            const boost::json::value * button_value_ptr = FireKey_object.if_contains("button");
+            if (button_value_ptr != nullptr) {
+                mouse.FireKey.button = boost::json::value_to<int>(*button_value_ptr);
+            }
+            
+            const boost::json::value * modifier_value_ptr = FireKey_object.if_contains("modifier");
+            if (modifier_value_ptr != nullptr) {
+                mouse.FireKey.modifier = boost::json::value_to<std::string>(*modifier_value_ptr);
+            }
+            
+        }
+
+        
+            const boost::json::value * inverse_x_value_ptr = mouse_object.if_contains("inverse_x");
+            if (inverse_x_value_ptr != nullptr) {
+                mouse.inverse_x = boost::json::value_to<bool>(*inverse_x_value_ptr);
+            }
+            
+            const boost::json::value * inverse_y_value_ptr = mouse_object.if_contains("inverse_y");
+            if (inverse_y_value_ptr != nullptr) {
+                mouse.inverse_y = boost::json::value_to<bool>(*inverse_y_value_ptr);
+            }
+            
+            const boost::json::value * comment_value_ptr = mouse_object.if_contains("comment");
+            if (comment_value_ptr != nullptr) {
+                mouse.comment = boost::json::value_to<std::string>(*comment_value_ptr);
+            }
+            
```

Comment:
This seems like a lot of repetition. Could we simplify this by having a few inline methods to get different types and handle the errors?

Something like the following pseudo-code:
```
bool getBooleanValue(jsonData, key, default) {
    value = jsonData[key]
    if value {
        return value;
    else {
        return default;
    }
}
```
though with optional range possible range limits (default to type max/min) for integers/floats/doubles as well?

Reply:
It is a lot of repetition. It's machine-generated code, generated by JsonStruct. As such, is it really a high priority to optimize it?

Reply:
hmm... we should make a note of that then; but perhaps JsonStruct needs something to make it more maintainable.

BTW, this is one reason I'm not too keen on code generators.